l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -477,4 +501,83 @@ contract ExecutorFacet is Base, IExecutor {
+    // Also verify that the provided commitment matches the provided versioned_hash.
+    function _pointEvaluationPrecompile(
+        bytes32 _versionedHash,
+        bytes32 _inputPoint,
+        bytes calldata _valueCommitmentProof
+    ) internal view {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _inputPoint, _valueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.staticcall(precompileInput);
+
+        // ToDo: Check output against spec, should have field elements prepended but other clients dont use it.
+        require(success, "failed to call point evaluation precompile");
+        (, uint256 result) = abi.decode(data, (uint256, uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    /// Verifies that the blobs contain the correct data by calling the point evaluation precompile. For the precompile we need:
+    /// versioned hash || opening point || opening value || commitment || proof
+    /// the pubdata will contain the last 3 values, the versioned hash is pulled from the BLOBHASH opcode, and the opening point
+    /// is calculated as the hash(versioned hash || blob commitment) where the blob commitment comes from system logs.
+    function _verifyBlobInformation(
+        bytes calldata _pubdataCommitments,
+        bytes32 _blob1Hash,
+        bytes32 _blob2Hash
+    ) internal view returns (bytes32[] memory blobCommitments) {
+        uint256 versionedHashIndex = 0;
+
+        // ToDo: This should be dynamic instead of being hardcoded to 2 blobs
+        require(_pubdataCommitments.length > 0, "pl");
+        require(
+            _pubdataCommitments.length <= PUBDATA_COMMITMENT_SIZE * 2 &&
+                _pubdataCommitments.length % PUBDATA_COMMITMENT_SIZE == 0,
+            "bs"
+        );
+        blobCommitments = new bytes32[](2);
+        bytes32 versionedHash;
+
+        for (uint256 i = 0; i < _pubdataCommitments.length; i += PUBDATA_COMMITMENT_SIZE) {
+            versionedHash = _getBlobVersionedHash(versionedHashIndex);
+
+            // ToDo: this can be made to be dynamic where evalutationPoint = keccak256(abi.encode(versionedHash, hashes[i / PUBDATA_COMMITMENT_SIZE]));
+            bytes32 evalutationPoint;
+            if (i < PUBDATA_COMMITMENT_SIZE) {
+                evalutationPoint = keccak256(abi.encode(versionedHash, _blob1Hash));
+            } else {
+                evalutationPoint = keccak256(abi.encode(versionedHash, _blob2Hash));
+            }
+
+            require(versionedHash != bytes32(0), "vh");
+
+            _pointEvaluationPrecompile(
+                versionedHash,
+                evalutationPoint,
+                _pubdataCommitments[i:i + PUBDATA_COMMITMENT_SIZE]
+            );
+
+            blobCommitments[versionedHashIndex] = keccak256(
+                abi.encodePacked(versionedHash, evalutationPoint, _pubdataCommitments[i:i + 32])
```

Comment:
I am fine with comments to save gas, esp since we don't do any manipulations with this array, just wanted to make sure that whenever person is reading the code, they can immediately understand the logic